### DISCUSSION TITLE
Image retrieval

### DISCUSSION THREAD

**Discussion by Zumbalamambo** [kornia/kornia#878] [Questions] [answered]
Thank you for the awesome local feature extractors. However which of them is most suitable for large scale image retrieval based tasks?

**Comment by ducha-aiki** [comment]
Unfortunately, none. AffNet+HardNet (both in kornia) work well via the classical bag-of-words model https://arxiv.org/abs/1711.06704
However, you would be much better off using a global descriptor like GeM. 
We also plan to include DELF https://github.com/nashory/DeLF-pytorch, which works well for the retrieval, but the global descriptor-based systems are much simpler and as good as local.
So I recommend you this one:

https://github.com/filipradenovic/cnnimageretrieval-pytorch

**Comment by edgarriba** [comment, marked as answer]
I opened this feature request ticket https://github.com/kornia/kornia/issues/880. @Zumbalamambo any help es very welcomed :)
